apps/namadillo/src/hooks/useSdk.tsx

```diff
@@ -12,7 +12,12 @@ const initializeSdk = async (): Promise<Sdk> => {
   const store = createStore();
   const rpcUrl = store.get(rpcUrlAtom);
   const nativeToken = store.get(nativeTokenAddressAtom);
-  const sdk = getSdk(cryptoMemory, rpcUrl, "", nativeToken);
+
+  if (!nativeToken.isSuccess) {
```

Comment:
I'm not sure if this is the best approach for sdk initialization, but now the native token is queried in chain parameters atom, not loaded from a .env file. Ideas?

Reply:
I guess the problem here is we want the SDK instance to depend on the native token and the RPC URL, which both come from atoms, but `initializeSdk` is just a normal function.

What about putting the global SDK instance inside an atom `sdkAtom` and replacing the calls to `getSdkInstance` with `useAtomValue(sdkAtom)` where possible (which should be possible for most cases)? And for whenever we need to access the SDK outside of React, we can change the definition of `getSdkInstance` to `store.get(sdkAtom)`, and callers of that function just need to be careful to make sure they are not getting an old SDK if the RPC/native token has changed.

If we do that, it should allow us to write the SDK atom to wait on `nativeTokenAddressAtom` being ready, and will also solve any problems of getting an old SDK instance inside our query atoms.

Alternatively, `initializeSdk` could return a Promise that resolves only when the native token is ready.

What do you think?